### ISSUE TITLE
Training set does not reflect test set

### ISSUE THREAD

**Issue by normancode** [georgesung/ssd_tensorflow_traffic_sign_detection#12]
I think you can remove this line:
`
# Only want data points where we have matching default boxes
if match_counter > 0:
    data_prep[image_file] = {'y_true_conf': y_true_conf, 'y_true_loc': y_true_loc}
`
Otherwise, the model assumes that all input are expected to have an object. Depending on the application this may not hold true. Train set should really reflect distribution in test set and that includes images without any object
